### DISCUSSION TITLE
Feature request, Add to_json/from_json to align with other to/from binary api.

### DISCUSSION THREAD

**Discussion by lygstate** [nlohmann/json#3068] [Q&A]
I wanna do something like this:
```
    switch (serialize_type) {
    case PreonCapiSerializeJSON:
        j = nlohmann::json::parse(json_data, json_data + json_size);
        break;
    case PreonCapiSerializeCBOR:
        j = nlohmann::json::from_cbor(json_data, json_data + json_size);
        break;
    case PreonCapiSerializeMessagePack:
        j = nlohmann::json::from_msgpack(json_data, json_data + json_size);
        break;
    case PreonCapiSerializeUBJSON:
        j = nlohmann::json::from_ubjson(json_data, json_data + json_size);
        break;
    case PreonCapiSerializeBSON:
        j = nlohmann::json::from_bson(json_data, json_data + json_size);
        break;
    default:
        return -101;
    }
```
But the json API stops me doing that.

**Comment by nlohmann** [comment]
In order to asses your issue, we need the following information:

- What is the issue you have?

- Please describe the steps to reproduce the issue. Can you provide a small but working code example?

- What is the expected behavior?

- And what is the actual behavior instead?

- Which compiler and operating system are you using? Is it a [supported compiler](https://github.com/nlohmann/json#supported-compilers)?

- Did you use a released version of the library or the version from the `develop` branch?

- If you experience a compilation error: can you [compile and run the unit tests](https://github.com/nlohmann/json#execute-unit-tests)?
